### PR TITLE
Change search order of ``sphinx.ext.inheritance_diagram``

### PR DESCRIPTION
**Pull request by tk0miya**
More details, please refer #3164.

### CODE REVIEW

File: tests/test_ext_inheritance_diagram.py [hunk 1]
```diff
@@ -9,9 +9,57 @@
...
+        # ignore current module if name include the module name
+        classes = import_classes('i18n.CatalogInfo', 'sphinx.util')
+        assert classes == [CatalogInfo]
+
+        # got exception for functions
+        raises(InheritanceException, import_classes, 'encode_uri', 'sphinx.util')
+
+        # try to load example.sphinx, but inheritance_diagram imports sphinx instead
+        # refs: #3164
+        classes = import_classes('sphinx', 'example')
```
Comment (tk0miya) [inline tests/test_ext_inheritance_diagram.py:61]: So far, inheritance_diagram returns `sphinx` package instead `example.sphinx` (if found).
This is incompatible change. But I think this is a unexpected behavior. in other words, it's bug.

File: tests/test_ext_inheritance_diagram.py [hunk 2]
```diff
@@ -9,9 +9,57 @@
...
+        # specified class in the module
+        classes = import_classes('sphinx.application.Sphinx', None)
+        assert classes == [Sphinx]
+
+        # specified class in current module
+        classes = import_classes('Sphinx', 'sphinx.application')
+        assert classes == [Sphinx]
+
+        # ignore current module if name include the module name
+        classes = import_classes('i18n.CatalogInfo', 'sphinx.util')
```
Comment (tk0miya) [inline tests/test_ext_inheritance_diagram.py:54]: This is also changed. So far, inheritance_diagram try to import `i18n.CatalogInfo` and always ignores `sphinx.util`; current module settings.

File: tests/test_ext_inheritance_diagram.py [hunk 3]
```diff
@@ -9,9 +9,57 @@
...
+
+        # specified class in the module
+        classes = import_classes('sphinx.application.Sphinx', None)
+        assert classes == [Sphinx]
+
+        # specified class in current module
+        classes = import_classes('Sphinx', 'sphinx.application')
+        assert classes == [Sphinx]
+
+        # ignore current module if name include the module name
```
Comment (shimizukawa) [inline tests/test_ext_inheritance_diagram.py]: Is this comment correct?
In this case, `sphinx.util.i18n` module will be imported at [sphinx/ext/inheritance_diagram.py L81](https://github.com/sphinx-doc/sphinx/pull/3171/files#diff-502071888c0340c81f31c9f00c7ec5a5R81) , so current module 'sphinx.util' is not ignored.

**Comment by shimizukawa** [comment]
LGTM with nits.

File: tests/test_ext_inheritance_diagram.py (hunk 3 above)
Comment (tk0miya) [reply tests/test_ext_inheritance_diagram.py]: ah, sorry.
I'd forgotten to rewrite the comment when I changed the search order.
It describes *old* behavior. not new one.

**Pull request merged**